Zhao1

eastnile opened the pull request:
1. updated package to use tidycensus instead of ACS to pull census data. Main changes are in acs.R
2. updated corresponding documentation

eastnile committed:
update to remove dependency on acs package

The ACS package was removed from CRAN. Update all functions to use tidycensus to query ACS data instead.

eastnile committed:
updated documentation

- updated docs
- checked for errors
- changed state and county choropleth ACS functions to pass to state_choropleth and county_choropleth via "..."

arilamstein commented on DESCRIPTION:
```diff
@@ -31,15 +31,14 @@ Imports:
 Depends:
     R (>= 3.5.0),
-    acs
+    tidycensus
```
The package should list tidycensus in Imports, not Depends. There is a long reason why acs was moved to Depends. I don't remember all the details, but I think at one point acs made a change, and I could perhaps no longer have package code which accessed its S4 objects unless the package was in Depends.

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
-#' will use a continuous scale. A value in [2, 9] will use that many colors. 
-#' @param zoom An optional list of states to zoom in on. Must come from the "name" column in
-#' ?state.regions.
+#' Creates a choropleth of US counties using the US Census' American Community Survey (ACS) data.  
```
I believe that "counties" here is a typo. The function below is state_choropleth_acs.

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
-#' @param zoom An optional list of states to zoom in on. Must come from the "name" column in
-#' ?state.regions.
+#' Creates a choropleth of US counties using the US Census' American Community Survey (ACS) data.  
+#' Requires the tidycensus package to be installed.
```
This is superfluous. By installing the package the tidycensus package is automatically installed. That's what it means to have tidycensus listed in any part of the DESCRIPTION file except "Suggests".

arilamstein replied:
Interesting, I just now see that I wrote a similar comment in `get_acs_data`. I was wrong to do that back then. I am not sure why I wrote that then, but I assume it is because this was my first package, the function started off as a standalone script, and I simply did not know better.

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
-#' }
-#' @importFrom acs acs.fetch geography estimate geo.make
-state_choropleth_acs = function(tableId, endyear=2011, span=5, num_colors=7, zoom=NULL)
+#' # Create a state choropleth for median household income*zooming in on New York, New Jersey and Connecticut
```
Is the * here a typo? The version it replaces uses a comma, which seems more standard.

eastnile replied:
Fixed.

arilamstein replied:
While you fixed the typo, it appears that there is another issue. See the attached screenshot. Line 24 begins with `#' on`. But because it is a comment, it should begin with `#' # on`.
<img width="744" alt="Screenshot 2025-03-19 at 8 49 59 AM" src="https://github.com/user-attachments/assets/9d85a900-a9ee-4b89-89fb-9c3ca3e805ac" />

In general, I recommend reviewing all the documentation by hand by typing `?<function name>`. Because there is a compilation step involved in R package documentation, it's easy for things to go awry.

arilamstein commented on DESCRIPTION:
```diff
@@ -54,12 +53,10 @@ Collate:
     'tract.R'
     'utils.R'
-RoxygenNote: 7.2.3
+RoxygenNote: 7.3.2
```
Also: 
  * The version of the package should be updated to 4.0.0
  * You should be added as an author
  * You should be listed as the maintainer

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
-#' @importFrom acs acs.fetch geography estimate geo.make
-state_choropleth_acs = function(tableId, endyear=2011, span=5, num_colors=7, zoom=NULL)
+#' # Create a state choropleth for median household income*zooming in on New York, New Jersey and Connecticut
+#'state_choropleth_acs(variable = "B19013_001", endyear = 2011, num_colors=1, 
```
While this works, it is customary to put a space after the `'`. I.e. it should read `#' state_choropleth_acs(` instead of `#'state_choropleth_acs`. 

The same is true for the two lines below.

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
-#' match the names of states as they appear in the "region" column of ?state.regions.
-#' @param county_zoom An optional vector of counties to zoom in on. Elements of this vector must exactly 
-#' match the names of counties as they appear in the "region" column of ?county.regions.
+#' Creates a choropleth of US States using the US Census' American Community Survey (ACS) data.  
```
I think there's a typo here. It says "US States" but the function below is `county_choropleth_acs`.

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
-#' @param county_zoom An optional vector of counties to zoom in on. Elements of this vector must exactly 
-#' match the names of counties as they appear in the "region" column of ?county.regions.
+#' Creates a choropleth of US States using the US Census' American Community Survey (ACS) data.  
+#' Requires the tidycensus package to be installed.
```
Again, this is superfluous because it is not possible to install choroplethr without tidycensus.

eastnile replied:
Fixed.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
+}
+
+
+process_acs_request = function(variable, tableId , geography, year, span, census_api_key) {
```
Even though this function is not exported it should still follow the convention of explicitly importing all functions from external packages it uses. For example:

```
#' @importFrom tidycensus load_variables get_acs  # Add any other functions
proocess_acs_request
```

This might seem pedantic but it's not. It's defensive. If, in the future, all other import calls of those functions are removed then this function will fail.

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
+    if(!variable %in% allvars$name) {
+      stop(paste0('The requested variable was not found in the ', dataset, ' dataset.'))
+    } 
+    acs_df = as.data.frame(tidycensus::get_acs(geography = geography, variable = variable, year = year, cache_table = F))
```
Please use `TRUE` instead of `T` and `FALSE` instead of `F`. This codebase was written prior to the Tidyverse Style Guide being published, but I try to follow it in my R code now. https://style.tidyverse.org/syntax.html#logical-vectors

eastnile replied:
Fixed

arilamstein commented on R/acs.R:
```diff
@@ -4,88 +4,149 @@ if (base::getRversion() >= "2.15.1") {
+
+
 #' Returns a list representing American Community Survey (ACS) estimates
+#' THIS FUNCTION IS DEPRECIATED; USE TIDYCENSUS TO QUERY ACS DATA INSTEAD
```
Unfortunately, deprecating this function is complicated.

 I just checked and it is part of the syllabus in the course [Mapping Census Bureau Data in R with Choroplethr](https://www.census.gov/data/academy/courses/choroplethr.html) which is on the Census website. See the slides for Module 4 (search for the function name):

https://www2.census.gov/about/training-workshops/courses/choroplethr/module-4-presentation.pdf

I see two posibilites:
  1. We can rewrite the function to just call `process_acs_request`, or some other tidycensus code. So people will call the function just as they did before, and get the same result as before. But the way the function does it will change. From the perspective of the continuity, this is probably the best solution.
  2. We can depreciate the function. That course is over 10 years old, and things change, so it's probably Ok to drop the function for another. But the way we deprecate it, I believe, should be different than what you've done below. There is a package called `lifecycle` and it has a function called `deprecate_hard` that we should probably look into: https://lifecycle.r-lib.org/reference/deprecate_soft.html.

Note that some of the issues that R CMD Check is finding relates to us removing the `acs` package from Depends, but still calling some of the functions from the package here.

eastnile replied:
I think the solution is just to change process_acs_request to be named get_acs_data and match the corresponding arguments. The function even has the same return structure; I'm not sure why it didn't occur to me in the first place.

arilamstein commented on R/acs.R:
```diff
@@ -170,13 +235,14 @@ make_geo = function(map)
   }
 }
 
+# DEPRECIATED
```
Note that this function is not exported. So I recommend completely deleting it from the file. According to the comment it has code which is specific to the acs package.

eastnile replied:
Fixed

arilamstein commented on R/acs.R:
```diff
@@ -144,6 +207,7 @@ get_tract_acs_data = function(tracts, tableId, endyear=2012, span=5, column_idx=
   list(df=df, title=title) # need to return 2 values here
 }
 
+# DEPRECIATED 
```
Ditto. It's not exported, and is only useful for the acs acs package. So please just delete it.

eastnile replied:
Fixed

arilamstein commented on R/acs.R:
```diff
@@ -158,6 +222,7 @@ get_column_idx = function(acs.data, tableId)
   column_idx
 }
 
+# DEPRECIATED
```
Please delete this function.

eastnile replied:
Fixed

arilamstein commented on dev/sandbox.R:
```diff
@@ -0,0 +1,33 @@
+library(choroplethrMaps)
```
I feel like this file should not be committed in the main branch. I have not be actively following R package development for many years. Is it now common to have a `dev` directory?

eastnile replied:
According to our AI overlords this is an accepted practice. For me it's useful to have a place to put some scratch files.

arilamstein replied:
Can you provide a link to that? Locally I often have a `scratch.R` file, but I add it to my .gitignore file so it's never checked in. My concern is that there seems to be no need to share this code with others e.g. via CRAN. Or does the  submission process simply skip the contents of this directory when zipping up the package?

arilamstein commented on R/get_usa_demographics.R:
```diff
@@ -0,0 +1,101 @@
+#' # get some demographic data on US states from the 2010 5-year ACS
+#' df = get_state_demographics(endyear=2010, span=5)
+#' 
+#' # a choropleth map shows the location of the values
```
Using the Tidyverse Style Guide, multi-sentence comments should use sentence case (i.e. start with a capital letter) and also have a period. https://style.tidyverse.org/functions.html#comments

arilamstein reviewed (commented):
This is a huge improvement over what was previously there. That being said:
  * There are some things that must be changed. For example CRAN will simply not accept the package unless it passes R CMD check with no ERRORs, WARNINGs or NOTEs). 
  * Some things that should be changed. For example, we should delete all the functions which only do things with data from the acs package (as opposed to adding a "Deprecated" comment above it). Let's also try to follow the Tidyverse Style Guide so collaboration is easier. (Note that this package was started prior to the Tidyverse Style Guide being created, so you will see deviations from it). 
  * There are also some open questions that I think we will need to resolve before submitting to CRAN. I don't have the right answer for right now, but don't think they need to hold up progress on the issues, so I am leaving this review now. For example: at the very least we will want to know exactly what parts of the [choroplethr course](https://www.census.gov/data/academy/courses/choroplethr.html) that is hosted on the Census website will be broken by this PR, and what workaround we want to suggest. We can then provide Census with that information so they can update the copy on the website with an errata. Note that I'm not saying that we can't introduce breaking changes. But we need to proactively communicate them to Census.

eastnile commented:
Just to clarify, in order to make these corrections, I should change the code in my branch then simply submit a new pull request, correct?

arilamstein commented:
You should absolutely commit the changes to your branch.

I believe that the PR will auto-update with those changes. But please
check. And if they don't, then simply close the PR and open a new one.

On Tue, Mar 18, 2025 at 11:24 AM Zhaochen He ***@***.***>
wrote:

> Just to clarify, in order to make these corrections, I should change the
> code in my branch then simply submit a new pull request, correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/arilamstein/choroplethr/pull/2#issuecomment-2734324350>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAMMYEUISGCT7UUW3B7PMT2VBQHVAVCNFSM6AAAAABZHEIGTCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZUGMZDIMZVGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
> [image: eastnile]*eastnile* left a comment (arilamstein/choroplethr#2)
> <https://github.com/arilamstein/choroplethr/pull/2#issuecomment-2734324350>
>
> Just to clarify, in order to make these corrections, I should change the
> code in my branch then simply submit a new pull request, correct?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/arilamstein/choroplethr/pull/2#issuecomment-2734324350>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAMMYEUISGCT7UUW3B7PMT2VBQHVAVCNFSM6AAAAABZHEIGTCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMZUGMZDIMZVGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

arilamstein commented:
Another issue: When I type `library(roxygen2); roxygenize()` it modifies 3 files, including 2 of the .Rd files. This leads me to think that the PR has changes to the documentation that were not processed. Please make sure to run `roxygenize()` prior to the next commit.

Note that roxygenize() issues a lot of complaints about the R6 classes note being documented. These warnings can be ignored.

eastnile committed:
process_acs_request() reworked to get_acs_data()

1. process_acs_request() renamed to get_acs_data() and augments of the function modified for backward compat
2. Updated author/contributor info
3. Fixed some typos

arilamstein commented:
While there are other issues that should be addressed (or at least considered), I think that this is a substantial improvement to the codebase and so I will merge it now.
